lib/jormungandr/src/Cardano/Wallet/Jormungandr/Transaction.hs

```diff
@@ -78,22 +76,20 @@ newTransactionLayer
     -> TransactionLayer t k
 newTransactionLayer (Hash block0H) = TransactionLayer
     { mkStdTx = \keyFrom rnps outs -> do
-        -- NOTE
-        -- For signing, we need to embed a hash of the transaction data
-        -- without the witnesses (since we don't yet have them!). In this sense,
-        -- this is a transaction id as Byron nodes or the http-bridge
-        -- defines them.
-        let inps = fmap (second coin) rnps
-        wits <- forM rnps $ \(_, TxOut addr _) -> do
-            xprv <- maybeToRight (ErrKeyNotFoundForAddress addr) (keyFrom addr)
-            let payload = block0H <> getHash (signData inps outs)
-            pure $ mkTxWitness (fst xprv) (sign payload xprv)
-        let tx = Tx
-                { txId = fragmentId inps outs wits
-                , resolvedInputs = inps
-                , outputs = outs
-                }
-        return (tx, wits)
+        (_fragId, frag) <- constructTransactionFragment
+            rnps
+            outs
+            (\addr (Hash txHash) -> do
+                xprv <- maybeToRight (ErrKeyNotFoundForAddress addr) (keyFrom addr)
+                let payload = block0H <> txHash
+                pure $ mkTxWitness (fst xprv) (sign payload xprv)
+            )
+        -- TODO: Either 1) get witnesses from constructTransactionFragment
+        -- or 2) change the return type mkStdTx to (Tx, SerialisedTxBinary)
+        let msg = runGet getMessage $ BL.fromStrict frag
+        case msg of
```

Comment:
`constructGenericSignedTx` is simplifying things precisely because it disregards the possibility of intermediate results (like `(payload, ins, outs, wits, auth)`). It does the witness and payload signing while producing the final binary.

This is at odds with our use of `(Tx, [TxWitness])` as an serialisable intermediary in core.

I'd _think_ we could replace it with `(Tx, SignedTxBinary)` [thread](https://input-output-rnd.slack.com/archives/GBT05825V/p1573652690032900)